operator/src/main/resources/scripts/modelInImage.sh

```diff
@@ -849,6 +929,126 @@ function wdtUpdateModelDomain() {
   trace "Exiting wdtUpdateModelDomain"
 }
 
+function wdtHandleOnlineUpdate() {
+
+  trace "Entering wdtHandleOnlineUpdate"
+  # wdt shell script may return non-zero code if trap is on, then it will go to trap instead
+  # temporarily disable it
+  stop_trap
+  if [ -z ${MII_USE_ONLINE_UPDATE} ] || [ "false" == "${MII_USE_ONLINE_UPDATE}" ] \
+    || [ ! -f "/tmp/diffed_model.json" ] ; then
+    trace "Not using online update because of onlineUpdate.enabled=false(or not set) or no diffed model found"
+    return
+  fi
+
+  if [ "$(cat /tmp/diffed_model.json)" == "{}" ] ; then
+    trace "No difference in the models"
+    return
+  fi
+
+  # We need to extract all the archives, WDT online checks for file existence
+  # even for delete
+  #
+  mkdir -p ${DOMAIN_HOME}/lib || exitOrLoop
+  for file in $(sort_files ${IMG_ARCHIVES_ROOTDIR} "*.zip")
+    do
+        # expand the archive domain libraries to the domain lib
+        cd ${DOMAIN_HOME}/lib || return exitOrLoop
+        ${JAVA_HOME}/bin/jar xf ${IMG_ARCHIVES_ROOTDIR}/${file} wlsdeploy/domainLibraries/
+
+        if [ $? -ne 0 ] ; then
+          trace SEVERE  "Domain Source Type is FromModel, error in extracting domain libs ${IMG_ARCHIVES_ROOTDIR}/${file}"
+          exitOrLoop
+        fi
+
+        # expand the archive apps and shared lib to the wlsdeploy/* directories
+        # the config.xml is referencing them from that path
+
+        cd ${DOMAIN_HOME} || exitOrLoop
+        ${JAVA_HOME}/bin/jar xf ${IMG_ARCHIVES_ROOTDIR}/${file} wlsdeploy/
+
+        if [ $? -ne 0 ] ; then
+          trace SEVERE "Domain Source Type is FromModel, error in extracting application archive ${IMG_ARCHIVES_ROOTDIR}/${file}"
+          exitOrLoop
+        fi
+
+
+    done
+
+
+  # Save off the encrypted model
+  cp ${DOMAIN_HOME}/wlsdeploy/domain_model.json /tmp/encrypted_merge_model.json
+  local admin_user=$(cat /weblogic-operator/secrets/username)
+  local admin_pwd=$(cat /weblogic-operator/secrets/password)
+
+  local ROLLBACK_FLAG=""
+  if [ ! -z "${MII_CANCEL_CHANGES_IFRESTART_REQ}" ] && [ "${MII_CANCEL_CHANGES_IFRESTART_REQ}" == "true" ]; then
+      ROLLBACK_FLAG="-cancel_changes_if_restart_required"
+  fi
+  # DEBUG only
+  #cat /tmp/diffed_model.yaml
+
+  if [ -z ${AS_SERVICE_NAME} ] || [ -z ${ADMIN_PORT} ] ; then
+    trace SEVERE "Cannot find admin service name or port"
+    exitOrLoop
+  fi
+
+  local admin_url
+  if [ -z "${ADMIN_PORT_SECURE}" ] ; then
+    admin_url="t3://${AS_SERVICE_NAME}:${ADMIN_PORT}"
+  else
+    admin_url="t3s://${AS_SERVICE_NAME}:${ADMIN_PORT}"
```

Comment:
If t3s needs a cert, how do we get it to the introspector job?

Reply:
Hmm, it will be a problem for self signed cert... user probably have to specify the location of the truststore...

Reply:
or update the default trust store

Reply:
e.g. set  -Dweblogic.security.TrustKeyStore=DemoTrust in WLSDEPLOY_PROPERTIES in the domain spec.  We need to document this

Reply:
TBD Tom: File JIRA to have central place to document this if it's not already documented (value can be DemoTrust or an actual file system location), work with Johnny & Quang to get initial wording. Also need the same "-D" in JAVA_OPTIONS so that managed servers can use SSL to the admin server.  How does operator get a cert to talk to admin port for server health?  How about for ready checks?

Reply:
I will file a research/doc JIRA that's independent of MII.